pkg/cmd/cli/cmd/create/imagestream.go

```diff
@@ -0,0 +1,131 @@
+package create
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+const (
+	ImageStreamRecommendedName = "imagestream"
+
+	imageStreamLong = `
+Create a new image stream
+
+Image streams allow you to track, tag, and import images from other registries. They also define an
+access controlled destination that you can push images to.`
+
+	imageStreamExample = `  # Create a new image stream
+  %[1]s mysql`
+)
+
+type CreateImageStreamOptions struct {
+	IS     *imageapi.ImageStream
+	Client client.ImageStreamsNamespacer
+
+	Mapper       meta.RESTMapper
+	OutputFormat string
+	Out          io.Writer
+	Printer      ObjectPrinter
+}
+
+// NewCmdCreateImageStream is a macro command to create a new image stream
+func NewCmdCreateImageStream(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	o := &CreateImageStreamOptions{Out: out}
+
+	cmd := &cobra.Command{
+		Use:     name + " NAME",
+		Short:   "Create a new empty image stream.",
+		Long:    imageStreamLong,
+		Example: fmt.Sprintf(imageStreamExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(cmd, f, args))
+			cmdutil.CheckErr(o.Validate())
+			cmdutil.CheckErr(o.Run())
+		},
+		Aliases: []string{"is"},
+	}
+
+	cmdutil.AddOutputFlagsForMutation(cmd)
+	return cmd
+}
+
+func (o *CreateImageStreamOptions) Complete(cmd *cobra.Command, f *clientcmd.Factory, args []string) error {
+	o.IS = &imageapi.ImageStream{
+		ObjectMeta: kapi.ObjectMeta{},
+		Spec:       imageapi.ImageStreamSpec{},
+	}
+
+	switch len(args) {
+	case 0:
+		return fmt.Errorf("image stream name is required")
+	case 1:
```

Comment:
Can we at least have `--image` or something like that, which will create a non-empty ImageStream?